frontend/src/components/Song.js

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Text, Text3D } from "@react-three/drei";
+import { animated, useSpring } from "@react-spring/three";
+import { v4 as uuidv4 } from "uuid";
+
+import data from "../utils";
+const toneArmOnSoundeffect = new Audio("/soundeffects/tonearm_on_sound.mp3");
+toneArmOnSoundeffect.volume = 0.4;
+const vinylSoundeffect = new Audio("/soundeffects/vinyl_soundeffect.mp3");
+vinylSoundeffect.volume = 1;
+vinylSoundeffect.loop = true;
+
+export default function Song({
+  playing,
+  setPlaying,
+  toneArmFinished,
+  soundOn,
+  songIndex,
+}) {
+  const x = -3.2;
+  const y = 1.4;
+  const z = 4;
+
+  const [songs, setSongs] = useState(data()); // List of songs
+  const [currentSong, setCurrentSong] = useState({
+    name: "Beaver Creek",
+    cover:
+      "https://chillhop.com/wp-content/uploads/2020/09/0255e8b8c74c90d4a27c594b3452b2daafae608d-1024x1024.jpg",
+    artist: "Aso, Middle School, Aviino",
+    audio: "Beaver Creek.mp3",
+    color: ["#205950", "#2ab3bf"],
+    id: uuidv4(),
+    active: true,
+  });
+
+  // Handle change in song
+  useEffect(() => {
+    console.log("CHANGING SONG");
+    currentAudio.pause();
+    setPlaying(false);
+    setCurrentSong(songs[songIndex]);
+  }, [songIndex]);
+
+  useEffect(() => {
+    console.log(currentSong);
+    if (currentSong["audio"]) {
+      setCurrentAudio(
+        new Audio(process.env.PUBLIC_URL + "/songs/" + currentSong["audio"])
+      );
+      console.log(currentAudio);
+    }
+  }, [currentSong]);
+
+  useEffect(() => {
+    setCurrentSong(songs[0]);
+  }, [songs]);
```

Comment:
I am not sure if this will ever get run. Basically if `songs` is ever changed, then it will run this effect, but I don't see anything that changes the song state.

https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
If the intention of this is to run it upon first render and it does the intended task then disregard this message

Reply:
It was indeed intended to run on the first render, especially because the songs list is initialized from a separate function. Sometimes the `currentSong` is loaded before the `songs` is loaded in, which crashes the app.

However, I do think the `useEffect` will need to change - `songs` is intended to be the Spotify queue of songs for the user. We don't want to set the current song to the first song every time a new song is added to `songs` (which will happen in the current code).

Reply:
I vote let's keep the `songs` / `setSongs` parts as is for now, we'll leave it as imperfect code snippets that exist as to-do's once the API is working.

Reply:
Okay, just keep a mental note to revisit this later.